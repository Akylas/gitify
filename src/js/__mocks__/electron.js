window.Notification = function (title, options) {
  this.title = title;

  return {
    onclick: jest.fn()
  };
};

window.Audio = class Audio {
  constructor(path) {
    this.path = path;
  }

  play() {}
};

window.localStorage = {
  store: {},
  getItem: function (key) { return this.store[key]; },
  setItem: function (key, item) {
    this.store[key] = item;
  },
  removeItem: jest.fn()
};

window.alert = jest.fn();

const browserWindow = {
  loadURL: jest.fn(),
  webContents: {
    on: () => {},
  },
  on: () => {},
  close: jest.fn(),
  destroy: jest.fn()
};

module.exports = {
  remote: {
    BrowserWindow: () => browserWindow
  },
  ipcRenderer: {
    send: jest.fn()
  },
  shell: {
    openExternal: jest.fn()
  },
};
