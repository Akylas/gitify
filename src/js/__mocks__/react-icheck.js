import React from 'react';

export class Checkbox extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="checkbox"
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class Radio extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="radio"
          value={this.props.value}
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class RadioGroup extends React.PureComponent {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  };
}
