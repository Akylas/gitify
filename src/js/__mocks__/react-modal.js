import React from 'react';

export default class Modal extends React.PureComponent {
  render() {
    return this.props.isOpen ? (
      <div>{this.props.children}</div>
    ) : null;
  };
}
