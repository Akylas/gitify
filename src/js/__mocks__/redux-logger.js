module.exports = function createLogger () {
  return ({ getState }) => (next) => (action) => {
    return next(action);
  };
};
