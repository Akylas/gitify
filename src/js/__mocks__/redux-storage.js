import { Map } from 'immutable';

export function createMiddleware(engine, actionBlacklist = [], actionWhitelist = []) {
  return ({ dispatch, getState }) => {
    return (next) => (action) => {
      return next(action);
    };
  };
};

export function createLoader() {
  const state = {
    auth: Map({
      token: 'LOGGED_IN_TOKEN'
    })
  };

  return () => new Promise(resolve => resolve(state));
}

export function reducer (clb) {
  return (state, action) => clb(state, action);
};
