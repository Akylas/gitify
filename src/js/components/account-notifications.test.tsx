import * as React from 'react';
import * as TestRendener from 'react-test-renderer';

import { AccountNotifications } from './account-notifications';
import { mockedGithubNotifications } from '../__mocks__/mockedData';

jest.mock('./repository');

describe('components/account-notifications.tsx', () => {
  it('should render itself (github.com with notifications)', () => {
    const props = {
      hostname: 'github.com',
      notifications: mockedGithubNotifications.toJS(),
    };

    const tree = TestRendener.create(<AccountNotifications {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render itself (github.com without notifications)', () => {
    const props = {
      hostname: 'github.com',
      notifications: [],
    };

    const tree = TestRendener.create(<AccountNotifications {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
