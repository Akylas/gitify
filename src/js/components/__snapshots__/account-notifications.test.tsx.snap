// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/account-notifications.tsx should render itself (github.com with notifications) 1`] = `
<div
  className="sc-bdVaJa dCCnan"
>
  github.com
  <svg
    aria-hidden="true"
    className="octicon"
    height={20}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 10 16"
    width={12.5}
  >
    <path
      d="M5 11L0 6l1.5-1.5L5 8.25 8.5 4.5 10 6l-5 5z"
      fillRule="evenodd"
    />
  </svg>
</div>
`;

exports[`components/account-notifications.tsx should render itself (github.com without notifications) 1`] = `
<div
  className="sc-bdVaJa dCCnan"
>
  github.com
  <svg
    aria-hidden="true"
    className="octicon"
    height={20}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "fill": "currentColor",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 8 16"
    width={10}
  >
    <path
      d="M5.5 3L7 4.5 3.25 8 7 11.5 5.5 13l-5-5 5-5z"
      fillRule="evenodd"
    />
  </svg>
</div>
`;
